contentcuration/contentcuration/frontend/channelList/views/KImg/index.vue

```diff
@@ -0,0 +1,404 @@
+<template>
+
+  <span :style="rootContainerStyles">
```

Comment:
Perhaps someone else worked on this, and it was simply vendored here. Anyways, the many `<span>` tags here aren't appropriate because the purpose of this is to display it as 'block-level content'. Using `<div>` will simplify the styles because `<span>` is for 'inline-level'

Reply:
Thanks @bjester  for noting this - I have addressed it on KDS

Reply:
I can understand that using `div` would simplify styles. Using `span`s was intentional approach. If `KImg` was implemented as `div` and then used within `span`, which is a valid use-case, then the resulting markup would be `<span><div>...</div></span>`.  Generally, placing block level elements within inline level elements is discouraged if not disallowed (I don't recall the source anymore, but technically it may actually be allowed in HTML5 though - still I wouldn't consider it very intuitive so why to do so).

More high-level, it is a part of an approach that separates HTML semantics from styling, which is a pattern I lean towards generally as often it helps with accessibility. It is true that in some cases it can add further work with styling.

No problem to discuss this further together @bjester, I just said @AllanOXDi that he doesn't need to be dealing with it in the scope of the card work.

Reply:
Yeah so instead of manipulating the styling, you can also manipulate the tag: `<component :is="...">`

This can be helpful for semantic structures, because lets say you code a generic component as a `<section>` but then you realize there's a case for using it as an `<article>` or an `<aside>`. All of those adhere to the same semantic structural scenarios in which to use `<header>` and `<footer>` as descendants.

Reply:
Yes that sounds like a useful technique for cases you need to switch element based on context